### PR TITLE
Nobel Prize-App

### PR DESCRIPTION
**Pull request by haliljon**
## In this PR I was able to:

- Use React documentation.
- Use React components.
- Use React props.
- Use React Router.
- Connect React and Redux.
- Handle events in a React app.
- Write integration tests with a React testing library.
- Use styles in a React app.
- Use React life cycle methods.
- Apply React best practices and language style guides in code.
- Use store, actions, and reducers in React.

### CODE REVIEW

File: src/pages/EachCategory.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  for (let i = 0; i < nobelPrizes.length; i += 1) {
+    if (nobelPrizes[i].category === params.categoryName) {
+      newArray.push(nobelPrizes[i]);
+    }
+  }
+  const [searchTerm, setSearchTerm] = useState([]);
+  const handleChange = (event) => {
+    setSearchTerm(event.target.value);
+  };
+  const filteredNewArray = newArray.filter((item) => item.year.toLowerCase().includes(searchTerm));
```
Comment (t-yanick) [inline src/pages/EachCategory.js:20]: - [ ] Great job working on this capstone project. Your category page is not displaying the list of noble price winners. When i try refreshing the page after clicking on a category, I get and error. See both screenshots below.
- **No list of noble prize winners diplaying for the medcine catgory**
Screen Shot 2023-02-15 at 4 47 27 AM
- **Error when I refresh the page**
Screen Shot 2023-02-15 at 4 45 02 AM

- Check the console as I am getting an error with the API GET method for fectching data for categories.
Screen Shot 2023-02-15 at 5 24 45 AM

File: src/tests/fetchPrizes.test.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import fetchPrizeData from '../services/fetchPrizes';
+
+describe('Fetch Data', () => {
+  test('gets the correct data', async () => {
+    const data = await fetchPrizeData();
+    expect(data).toBeInstanceOf(Object);
+    expect(data.id).not.toBeNull();
+  });
+});
```
Comment (t-yanick) [inline src/tests/fetchPrizes.test.js:9]: - [ ] Great job having a test file. But you need to add more tests, both unit and integration tests. See screenhot below of what you have to add for your testing to be as per project requirements.
Screen Shot 2023-02-15 at 5 11 22 AM

File: src/index.css [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  height: auto;
+  border: 0;
+  outline: 0;
+  padding: 0 4%;
+  border-radius: 30px;
+  text-align: center;
+}
+
+/* each category has a different background color */
+
```
Comment (t-yanick) [inline src/index.css]: - [ ] As per the Behance design in mobile view, there are 2 columns but your design has 3 columns. Kindly ensure that you design is as close as possible to the original design.
- **Original Design**
Screen Shot 2023-02-15 at 5 19 23 AM
- **Your Design**
Screen Shot 2023-02-15 at 5 20 24 AM

**Review by t-yanick — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️

Hi @haliljon , 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Project documented
:heavy_check_mark: Appropriate credit to Nelson Sakwa, author of the original design used for this project
:heavy_check_mark: App deployed on Netlify
:heavy_check_mark: Good page structure
:heavy_check_mark: Use of Redux for state management

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️

- [ ] Good job doing a video presentation of the app, but you are expected to include as part of your presentation a piece of code which you are proud of or find interesting. Kindly re-do the video presentation.
- [ ] Kindly remove the Create React App boilerplate files which you have not used from your project directory. This makes it clean and professional.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/pages/EachCategory.js (hunk 1 above)
Comment (haliljon) [reply src/pages/EachCategory.js:20]: Hi @t-yanick, I can checked it from my end with a different device and it is working pretty well. 
Here is the video recording of the website:
https://user-images.githubusercontent.com/110017001/219054667-c7126a6f-3f23-41b6-92e5-f4ce95e7be62.MOV

Comment (t-yanick) [reply src/pages/EachCategory.js:20]: - Kindly fix all the other issues and request for a re-review.

File: src/tests/Navbar.test.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+it('matches snapshot', () => {
+  const tree = renderer.create(
+    <Provider store={store}>
+      <Router>
+        <Navbar />
+      </Router>
+    </Provider>,
+  ).toJSON();
+  expect(tree).toMatchSnapshot();
+});
```
Comment (AnselemOdims) [inline src/tests/Navbar.test.js:17]: - [ ] At the moment, you only wrote snapshot tests. Please as pointed out by the previous reviewer, you need to include more snapshot/unit tests and integration tests.

**Review by AnselemOdims — CHANGES_REQUESTED** [review]
### Hi @haliljon , 
You have been doing a tremendous job so far :ok_hand:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
No linter errors :heavy_check_mark:

## Required Changes ♻️
- [ ] Please also check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by christianbiring1 — APPROVED** [review]
Hi @haliljon , 

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit:
 
# STATUS: APPROVED :tada: 
nicee

Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**